tools/partutil/extend_partition.go

```diff
@@ -0,0 +1,136 @@
+package partutil
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+//ExtendPartition extends a partition to a specific end sector
+func ExtendPartition(disk string, partNumInt, end int) error {
+	if len(disk) <= 0 || partNumInt <= 0 {
+		return errors.New("invalid disk name or partition number")
+	}
+
+	//for cases like loop5p1
+	partNum := strconv.Itoa(partNumInt)
+	if disk[len(disk)-1] >= '0' && disk[len(disk)-1] <= '9' {
+		partNum = "p" + partNum
+	}
+
+	partName := disk + partNum
+
+	//dump partition table to a file
+	cmd := string("sudo sfdisk --dump ") + disk + " > extend_partition_tmp"
+	err := ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+	defer os.Remove("extend_partition_tmp")
+
+	err = editPartitionTableFile("extend_partition_tmp", partName, end)
+	if Check(err, "editing partition table file of "+partName+" to ending sector at: "+strconv.Itoa(end)) {
+		return err
+	}
+
+	//write partition table back
+	cmd = "sudo sfdisk " + disk + " < " + " extend_partition_tmp "
+	err = ExecCmdToStdout(cmd)
+	if Check(err, cmd) {
+		return err
+	}
+	log.Printf("\nCompleted extending %s\n\n", partName)
+
+	// check and repair file system in the partition
+	cmd = "sudo e2fsck -fp " + partName
```

Comment:
As you are running all the commands with sudo mode, I think maybe exec_cmd_to_stdout can wrap that up.